GeosCore/fullchem_mod.F90

```diff
@@ -1006,14 +1009,34 @@ SUBROUTINE Do_FullChem( Input_Opt,  State_Chm, State_Diag,                 &
        !
        ! Archive KPP reaction rates [s-1]
        ! See gckpp_Monitor.F90 for a list of chemical reactions
+       !
+       ! NOTE: In KPP 2.5.0+, VAR and FIX are now private to the integrator
+       ! and point to C.  Therefore, pass C(1:NVAR) instead of VAR and
+       ! C(NVAR+1:NSPEC) instead of FIX to routine FUN.
        !=====================================================================
        IF ( State_Diag%Archive_RxnRate ) THEN
 #ifdef MODEL_GEOS
-          ! gckpp_Function.F90 function Fun must be modified to use optional
-          ! argument Vdotout if building with GEOS
-          CALL Fun( VAR, FIX, RCONST, Vloc, Aout=Aout, Vdotout=Vdotout )
+          !---------------------------------------------------
+          ! GEOS-Chem in NASA/GEOS:
+          ! Get equation rates (Aout) and the time derivative
+          ! of variable species concentrations (Vdotout)
+          !---------------------------------------------------
+          CALL Fun( V       = C(1:NVAR),                                     &
+                    F       = C(NVAR+1:NSPEC),                               &
+                    RCT     = RCONST,                                        &
+                    Vdot    = Vloc,                                          &
+                    Aout    = Aout,                                          &
+                    Vdotout = Vdotout                                       )
```

Comment:
KPP 2.5.0 now automatically adds Aout and Vdotout variables as optional arguments to subroutine Fun().  So the GMAO folks won't have to hack this in going forward.